x/programs/test/programs/simple_token/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use wasmlanche_sdk::Context;
+use wasmlanche_sdk::{public, state_keys, types::Address, Program};
+
+const INITIAL_SUPPLY: i64 = 123456789;
+
+/// The program state keys.
+#[state_keys]
+pub enum StateKey {
+    /// The total supply of the token. Key prefix 0x0.
+    TotalSupply,
+    /// The name of the token. Key prefix 0x1.
+    Name,
+    /// The symbol of the token. Key prefix 0x2.
+    Symbol,
+    /// The balance of the token by address. Key prefix 0x3 + address.
+    Balance(Address),
+
+    Allowance([Address; 2]),
+
+    Owner,
+}
+
+pub fn get_owner(program: &Program<StateKey>) -> Address {
+    program
+        .state()
+        .get(StateKey::Owner)
+        .expect("failed to get total supply")
+}
+
+pub fn owner_check(program: &Program<StateKey>, actor: Address) {
+    assert!(
+        get_owner(program) == actor,
+        "caller is required to be owner"
+    )
+}
+
+/// Initializes the program with a name, symbol, and total supply.
+#[public]
+pub fn init(context: Context<StateKey>, name: String, symbol: String, owner: Address) {
+    let Context { program, actor } = context;
+
+    program
+        .state()
+        .store(StateKey::Owner, &actor)
+        .expect("failed to store total supply");
+
+    // set total supply
+    program
+        .state()
+        .store(StateKey::TotalSupply, &INITIAL_SUPPLY)
+        .expect("failed to store total supply");
+
+    // set token name
+    program
+        .state()
+        .store(StateKey::Name, &name)
+        .expect("failed to store coin name");
+
+    // set token symbol
+    program
+        .state()
+        .store(StateKey::Symbol, &symbol)
+        .expect("failed to store symbol");
+}
+
+/// Returns the total supply of the token.
+#[public]
+pub fn total_supply(context: Context<StateKey>) -> i64 {
+    let Context { program, .. } = context;
+    program
+        .state()
+        .get(StateKey::TotalSupply)
+        .expect("failed to get total supply")
+}
+
+/// Transfers balance from the token owner to the recipient.
+#[public]
+pub fn mint(context: Context<StateKey>, recipient: Address, amount: i64) -> bool {
+    let Context { program, actor } = context;
+    owner_check(&program, actor);
+    let balance = program
+        .state()
+        .get::<i64>(StateKey::Balance(recipient))
```

Comment:
```suggestion
        .get::<u64>(StateKey::Balance(recipient))
```

I would probably use a type-alias at the top of the file `type BalanceType = u64;` and use it for everything. Not 100% sure name, maybe `QuantityType`, 🤷 .